url3472127e1f_SergeyKrivohatskiy_tiny_database_tgz-pJ8-ru_spbau_tinydb_bufferManager_IndexTestJ8/src/ru/spbau/tinydb/utils/Utils.java

```diff
@@ -0,0 +1,149 @@
+package ru.spbau.tinydb.utils;
+
+import ru.spbau.tinydb.btree.BxTreeEntry;
+import ru.spbau.tinydb.queries.Attribute;
+import ru.spbau.tinydb.queries.SecondLevelId;
+import ru.spbau.tinydb.table.Record;
+import ru.spbau.tinydb.table.Table;
+
+import java.io.UnsupportedEncodingException;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * tiny_database
+ * Created by Sergey on 08.11.2014.
+ */
+public class Utils {
+    public static byte[] intToBytes(int i) {
+        return ByteBuffer.allocate(4).putInt(i).array();
+    }
+
+    public static int bytesToInt(byte[] intBytes) {
+        return ByteBuffer.wrap(intBytes, 0, 4).getInt();
+    }
+
+    public static byte[] stringToBytes(String value, int size) throws UnsupportedEncodingException {
+        byte[] res = new byte[size];
+        byte[] stringBytes = value.getBytes("US-ASCII");
+        if(stringBytes.length > size) {
+            throw new IllegalArgumentException("String len is more than size");
+        }
+        System.arraycopy(stringBytes, 0, res, 0, stringBytes.length);
+        Arrays.fill(res, stringBytes.length, res.length, (byte) 0);
+        return res;
+    }
+
+    public static String bytesToString(byte[] bytes) {
+        try {
+            int i = 0;
+            for(; i < bytes.length; i++) {
+                if(bytes[i] == 0) {
+                    break;
+                }
+            }
+            return new String(bytes, 0, i, "US-ASCII");
+        } catch (UnsupportedEncodingException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)